### ISSUE TITLE
TFLite avg_pool2d returns -inf instead of NaN for pooling window containing NaN

### ISSUE THREAD

**Issue by ALinrunrun** [tensorflow/tensorflow#118388]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

2.21.0

### Custom code

Yes

### OS platform and distribution

Linux Ubuntu 22.04

### Python version

3.11

### Current behavior?

TensorFlow Lite produces a different result from Keras for tf.nn.avg_pool2d when the input contains NaN.

The model applies avg_pool2d with ksize=2, strides=2, and padding="VALID" to a 4x4 input. One value in the last pooling window is NaN. Keras propagates NaN for that output element, while the converted TFLite model returns -inf.

In the reproducer below, Keras returns [5, 5, 5, nan], while TFLite returns [5, 5, 5, -inf].

Expected behavior?

The converted TFLite model should match Keras behavior for tf.nn.avg_pool2d on inputs containing NaN. It should not convert a NaN pooling result into -inf.

### Standalone code to reproduce the issue

```shell
import os
os.environ["TF_CPP_MIN_LOG_LEVEL"] = "3"

import numpy as np
import tensorflow as tf


class M(tf.keras.Model):
    def call(self, x):
        return tf.nn.avg_pool2d(x, ksize=2, strides=2, padding="VALID")


model = M()
model.build((None, 4, 4, 1))

tflite_model = tf.lite.TFLiteConverter.from_keras_model(model).convert()

x = np.full((1, 4, 4, 1), 5.0, dtype=np.float32)
x[0, 2, 3, 0] = np.nan

keras_out = model(x).numpy().flatten()

interpreter = tf.lite.Interpreter(model_content=tflite_model)
interpreter.allocate_tensors()
interpreter.set_tensor(interpreter.get_input_details()[0]["index"], x)
interpreter.invoke()

tflite_out = interpreter.get_tensor(
    interpreter.get_output_details()[0]["index"]
).flatten()

print("eager :", keras_out)
print("tflite:", tflite_out)

if np.isnan(keras_out).any() and np.isneginf(tflite_out).any():
    print("BUG REPRODUCED: TFLite avg_pool2d returns -inf instead of NaN")
else:
    print("not reproduced")
```

### Relevant log output

```shell
eager : [ 5. 5. 5. nan]
tflite: [ 5. 5. 5. -inf]
BUG REPRODUCED: TFLite avg_pool2d returns -inf instead of NaN
```

**Comment by aravindmurali22** [comment]
Hi @ALinrunrun ,

Thank you for taking the time to report this and for your patience.

TFLite has [transitioned](https://developers.googleblog.com/en/tensorflow-lite-is-now-litert/) to LiteRT, Google's primary runtime for on-device AI. All active development, feature updates, and performance enhancements have shifted  to LiteRT. We encourage you to try running your model with LiteRT. If you still encounter this issue, please open a new issue with the updated details in the [LiteRT Repository](https://github.com/google-ai-edge/LiteRT).

For more information and step-by-step guidance on updating your project, please refer to the [LiteRT Migration Guide](https://developers.google.com/edge/litert/migration).
Thank you again!
